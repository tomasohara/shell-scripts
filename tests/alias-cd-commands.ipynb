{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "273ee5ab",
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "# Setup (OPTIONAL - USED IF NEEDED)\n",
    "## bind 'set enable-bracketed-paste off'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "90b1a7f3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PROMPT_BLYGGVEKNCRG\\[\\]>\n"
     ]
    }
   ],
   "source": [
    "## TEST:  Make sure simple prompt used (e.g., no escapes that might contaminate output)\n",
    "## PS1=\"$ \"\n",
    "## TODO: PS1=\"> \"\n",
    "\n",
    "## NOTE: The Jupyter bash kernel requires that PS1 not be modified as they customize it. \n",
    "echo $PS1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "c8665c2d-276b-4376-a666-0f1cf32bab91",
   "metadata": {},
   "outputs": [],
   "source": [
    "# # Delete all aliases and function\n",
    "# # TODO: Instead start from pristine environment\n",
    "# unalias -a\n",
    "# alias | wc -l\n",
    "# for f in $(typeset -f | egrep '^\\w+'); do unset -f $f; done\n",
    "# typeset -f | egrep '^\\w+' | wc -l"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "5e52c8d9",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Global Setup\n",
    "alias testnum=\"sed -r \"s/[0-9]/N/g\"\" \n",
    "alias testuser=\"sed -r \"s/\"$USER\"+/user/g\"\"\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "29cebb30",
   "metadata": {},
   "outputs": [],
   "source": [
    "## NOTE: For reproducability, the directory name needs to be fixed\n",
    "## In place of $$, use a psuedo random number (e,g., 6869)\n",
    "## *** All output from one run to the next needs to be the same ***\n",
    "\n",
    "## temp_dir=$TMP/test-$$\n",
    "## OLD\n",
    "# mkdir -p \"$temp_dir\"\n",
    "# # TODO: /bin/rm -rvf \"$temp_dir\"\n",
    "# cd \"$temp_dir\"\n",
    "# pwd\n",
    "\n",
    "TMP=/tmp/test-cp-mv\n",
    "temp_dir=$TMP/test-6869\n",
    "trash_dir=$TMP/\"_trash-$(date \"+%Y%m%d%H%M%S\")\"\n",
    "\n",
    "command mkdir -p $temp_dir\n",
    "command mkdir -p $trash_dir\n",
    "command cd $temp_dir\n",
    "\n",
    "## OLD\n",
    "# # ALIAS FOR PRINTING SEPERATION LINES (FOR JUPYTER)\n",
    "# alias linebr=\"printf '%*s\\n' \"${COLUMNS:-$(tput cols)}\" '' | tr ' ' -\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "9e1fb88a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "NNN\n"
     ]
    }
   ],
   "source": [
    "## OLD\n",
    "# Count aliases proper\n",
    "# alias | wc -l \n",
    "alias | wc -l | testnum"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "6a1cb00b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "NNN\n"
     ]
    }
   ],
   "source": [
    "## OLD\n",
    "# Count functions\n",
    "# typeset -f | egrep '^\\w+' | wc -l \n",
    "typeset -f | egrep '^\\w+' | wc -l | testnum"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "8a6085e9",
   "metadata": {},
   "outputs": [],
   "source": [
    "## BAD\n",
    "# source $BIN_DIR/tomohara-aliases.bash"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "6cce12a4",
   "metadata": {},
   "outputs": [],
   "source": [
    "## DEBUG\n",
    "## SHOWS CURRENT WORKING DIRECTORY\n",
    "# pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "069ed8cc",
   "metadata": {},
   "outputs": [],
   "source": [
    "## chdir works as cd\n",
    "chdir .. > /dev/null\n",
    "chdir test-6869/ > /dev/null"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "36fb8fbd",
   "metadata": {},
   "outputs": [],
   "source": [
    "## cd-this-realdir works as cd as well\n",
    "cd-this-realdir .. > /dev/null\n",
    "cd-this-realdir test-6869/ > /dev/null"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "4033a985",
   "metadata": {},
   "outputs": [],
   "source": [
    "## BAD: rm -rf ./* > /dev/null\n",
    "command mv -f * $trash_dir\n",
    "# mkdir testdir89 testdir90\n",
    "command mkdir testdir89 testdir90\n",
    "echo \"Testfile1\" > testdir89/f11.txt\n",
    "echo \"Testfile2\" > testdir89/f12.txt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "e4b7b6fc",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "f11.txt  f12.txt\n"
     ]
    }
   ],
   "source": [
    "pushd-q testdir89\n",
    "ls\n",
    "# linebr\n",
    "# popd-q \n",
    "# command ls"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "14b95b96",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "testdir89  testdir90\n"
     ]
    }
   ],
   "source": [
    "popd-q \n",
    "## OLD\n",
    "# ls\n",
    "command ls"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Bash",
   "language": "bash",
   "name": "bash"
  },
  "language_info": {
   "codemirror_mode": "shell",
   "file_extension": ".sh",
   "mimetype": "text/x-sh",
   "name": "bash"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
