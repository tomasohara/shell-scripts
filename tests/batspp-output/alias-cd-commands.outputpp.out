1..12
not ok 1 test-1
# (in test file batspp-output/alias-cd-commands.outputpp, line 25)
#   `[ "$(test-1-actual)" == "$(test-1-expected)" ]' failed
# ========== echo $PS1
#  ==========
#
# ========= [PEXP\[\]ECT_PROMPT> =========
# [PEXP\[\]ECT_PROMPT>
# ============================
not ok 2 test-3
# (in test file batspp-output/alias-cd-commands.outputpp, line 59)
#   `[ "$(test-3-actual)" == "$(test-3-expected)" ]' failed
# ========== unalias -a
#  ==========
# ========= $ alias | wc -l\n$ for f in ; do unset -f ; done\n$ typeset -f | egrep '^\w+' | wc -l\n0\n0 =========
# $ alias | wc -l
# $ for f in bats_abort_timeout_countdown ()
# bats_add_debug_exclude_path ()
# bats_all_in ()
# bats_any_in ()
# bats_append_arrays_as_args ()
# bats_approx_realpath ()
# bats_binary_search ()
# bats_capture_stack_trace ()
# bats_check_status_from_trap ()
# bats_check_valid_version ()
# bats_create_test_tmpdirs ()
# bats_debug_trap ()
# bats_emit_trace ()
# bats_error_trap ()
# bats_evaluate_preprocessed_source ()
# bats_execute ()
# bats_exit_trap ()
# bats_extract_line ()
# bats_format_file_line_reference ()
# bats_format_file_line_reference_colon ()
# bats_format_file_line_reference_comma_line ()
# bats_format_file_line_reference_uri ()
# bats_frame_filename ()
# bats_frame_function ()
# bats_frame_lineno ()
# bats_generate_warning ()
# bats_get_child_processes_of ()
# bats_get_failure_stack_trace ()
# bats_internal_load ()
# bats_interrupt_trap ()
# bats_interrupt_trap_in_run ()
# bats_kill_childprocesses_of ()
# bats_load_library ()
# bats_load_library_safe ()
# bats_load_safe ()
# bats_merge_stdout_and_stderr ()
# bats_normalize_windows_dir_path ()
# bats_perform_test ()
# bats_prefix_lines_for_tap_output ()
# bats_preprocess_source ()
# bats_print_failed_command ()
# bats_print_stack_trace ()
# bats_quote_code ()
# bats_readlinkf ()
# declare -fx bats_readlinkf
# bats_redirect_stderr_into_file ()
# bats_replace_filename ()
# bats_require_minimum_version ()
# bats_separate_lines ()
# bats_setup_tracing ()
# bats_should_retry_test ()
# bats_sort ()
# bats_start_timeout_countdown ()
# bats_strip_string ()
# bats_teardown_trap ()
# bats_test_begin ()
# bats_test_function ()
# bats_timeout_trap ()
# bats_trim ()
# bats_trim_filename ()
# bats_version_lt ()
# bats_warn_minimum_guaranteed_version ()
# find_in_bats_lib_path ()
# get_mills_since_epoch ()
# load ()
# run ()
# setup ()
# skip ()
# teardown ()
# test-1-actual ()
# test-1-expected ()
# END_EXPECTED
# test-10-actual ()
# test-10-expected ()
# END_EXPECTED
# test-11-actual ()
# test-11-expected ()
# END_EXPECTED
# test-12-actual ()
# test-12-expected ()
# END_EXPECTED
# test-13-actual ()
# test-13-expected ()
# f11.txt  f12.txt
# testdir89  testdir90
# END_EXPECTED
# test-3-actual ()
# test-3-expected ()
# 0
# 0
# END_EXPECTED
# test-4-actual ()
# test-4-expected ()
# 0
# END_EXPECTED
# test-5-actual ()
# test-5-expected ()
# END_EXPECTED
# test-6-actual ()
# test-6-expected ()
# 1
# END_EXPECTED
# test-7-actual ()
# test-7-expected ()
# 0
# END_EXPECTED
# test-8-actual ()
# test-8-expected ()
# END_EXPECTED
# test-9-actual ()
# test-9-expected ()
# END_EXPECTED
# test_test-2d1 ()
# test_test-2d10 ()
# test_test-2d11 ()
# test_test-2d12 ()
# test_test-2d13 ()
# f11.txt  f12.txt
# testdir89  testdir90' "=========";
# test_test-2d3 ()
# test_test-2d4 ()
# 0' "=========";
# test_test-2d5 ()
# test_test-2d6 ()
# test_test-2d7 ()
# test_test-2d8 ()
# test_test-2d9 () ; do unset -f ; done
# $ typeset -f | egrep '^\w+' | wc -l
# 0
# 0
# ============================
not ok 3 test-4
# (in test file batspp-output/alias-cd-commands.outputpp, line 97)
#   `[ "$(test-4-actual)" == "$(test-4-expected)" ]' failed
# ========== TMP=/tmp/test-cp-mv
#  ==========
# ========= ## NOTE: Source it directly from the ./tests directory.
# $ BIN_DIR=$PWD/..
# $ alias | wc -l
# 0 =========
# ## NOTE: Source it directly from the ./tests directory.
# $ BIN_DIR=/tmp/batspp-70711/test-4/..
# $ alias | wc -l
# 0
# ============================
not ok 4 test-5
# (in test file batspp-output/alias-cd-commands.outputpp, line 134)
#   `[ "$(test-5-actual)" == "$(test-5-expected)" ]' failed
# ========== temp_dir=$TMP/test-6869
#  ==========
# ========= $ mkdir -p "$temp_dir"
# # TODO: /bin/rm -rvf "$temp_dir"
# $ cd "$temp_dir"
# $ pwd
# #ALIAS FOR PRINTING SEPERATION LINES (FOR JUPYTER)
# $ alias linebr="printf '%*s\n' "${COLUMNS:-$(tput cols)}" '' | tr ' ' -"
# /tmp/test-cp-mv/test-6869 =========
# $ mkdir -p "/test-6869"
# # TODO: /bin/rm -rvf "/test-6869"
# $ cd "/test-6869"
# $ pwd
# #ALIAS FOR PRINTING SEPERATION LINES (FOR JUPYTER)
# $ alias linebr="printf '%*s\n' "80" '' | tr ' ' -"
# /tmp/test-cp-mv/test-6869
# ============================
not ok 5 test-6
# (in test file batspp-output/alias-cd-commands.outputpp, line 174)
#   `[ "$(test-6-actual)" == "$(test-6-expected)" ]' failed
# ========== alias | wc -l
#  ==========
# 0
# ========= 1 =========
# 1
# ============================
not ok 6 test-7
# (in test file batspp-output/alias-cd-commands.outputpp, line 208)
#   `[ "$(test-7-actual)" == "$(test-7-expected)" ]' failed
# ========== typeset -f | egrep '^\w+' | wc -l\n ==========
# 127
# ========= 0 =========
# 0
# ============================
not ok 7 test-8
# (from function `test-8-actual' in file batspp-output/alias-cd-commands.outputpp, line 249,
#  in test file batspp-output/alias-cd-commands.outputpp, line 236)
#   `test-8-actual ' failed
# ========== source $BIN_DIR/tomohara-aliases.bash
#  ==========
# /home/aveey/tom-project/shell-scripts/tests/batspp-output/alias-cd-commands.outputpp: line 249: /tomohara-aliases.bash: No such file or directory
not ok 8 test-9
# (in test file batspp-output/alias-cd-commands.outputpp, line 276)
#   `[ "$(test-9-actual)" == "$(test-9-expected)" ]' failed
# ========== pwd
#  ==========
# /tmp/batspp-70711/test-9
# ========= /tmp/test-cp-mv/test-6869 =========
# /tmp/test-cp-mv/test-6869
# ============================
not ok 9 test-10
# (from function `test-10-actual' in file batspp-output/alias-cd-commands.outputpp, line 317,
#  in test file batspp-output/alias-cd-commands.outputpp, line 304)
#   `test-10-actual ' failed with status 127
# ========== chdir ..
#  ==========
# /home/aveey/tom-project/shell-scripts/tests/batspp-output/alias-cd-commands.outputpp: line 317: chdir: command not found
not ok 10 test-11
# (from function `test-11-actual' in file batspp-output/alias-cd-commands.outputpp, line 353,
#  in test file batspp-output/alias-cd-commands.outputpp, line 340)
#   `test-11-actual ' failed with status 127
# ========== chdir ..
#  ==========
# /home/aveey/tom-project/shell-scripts/tests/batspp-output/alias-cd-commands.outputpp: line 353: chdir: command not found
not ok 11 test-12
# (in test file batspp-output/alias-cd-commands.outputpp, line 382)
#   `[ "$(test-12-actual)" == "$(test-12-expected)" ]' failed
# ========== rm -rf ./* > /dev/null
#  ==========
# ========= $ mkdir testdir89 testdir90
# $ echo "Testfile1" > testdir89/f11.txt
# $ echo "Testfile2" > testdir89/f12.txt =========
# $ mkdir testdir89 testdir90
# $ echo "Testfile1" > testdir89/f11.txt
# $ echo "Testfile2" > testdir89/f12.txt
# ============================
not ok 12 test-13
# (from function `test-13-actual' in file batspp-output/alias-cd-commands.outputpp, line 425,
#  in test file batspp-output/alias-cd-commands.outputpp, line 412)
#   `test-13-actual ' failed with status 127
# ========== pushd-q testdir89
#  ==========
# /home/aveey/tom-project/shell-scripts/tests/batspp-output/alias-cd-commands.outputpp: line 425: pushd-q: command not found
